### ISSUE TITLE
Glupen64 android arm7 build?

### ISSUE THREAD

**Issue by epayson85** [loganmc10/GLupeN64#46]
Any chance we can get a release of an android build? I have tried building myself and continue to get errors... (I will admit I am no expert at this haha so its most likely user error) I would love to test this out on my Shieldtv!

**Comment by loganmc10** [comment]
I literally just finished setting up a buildbot!

To get it on Android you need to change your retroarch.cfg

core_updater_buildbot_url = "http://loganbuildbot.s3-website-us-east-1.amazonaws.com/cores"

That setting is under "Settings > Updater > Buildbot Cores URL" in the menu

Once that is done just go to "Online Updater > Core Updater" and choose glupen64_android.zip. There is a gles3 version as well, but that won't work with RetroArch 1.3.6, you'll need a nightly, or wait for 1.3.7 to use it.

**Comment by loganmc10** [comment]
Unfortunately on Android you have to get the core through the Online Updater, on other platforms like Linux, Windows, Raspberry Pi, etc.... you can just copy the core (the .so/.dll file) into your "cores" folder.

Android has security measures in place to stop apps from executing "outside" code, and it doesn't allow you to use a shared library that isn't part of the app, so it won't let you put the file on your SD card and use it for instance.

The shared library has to "come from the app", thus the need to get it through the Online Updater. Anyway just an FYI for anyone who happens to read this in the future.

**Comment by epayson85** [comment]
Luckily I use hyperspin as a front end so it can launch the games using the core :) I just tried the gles3 version with latest nightly! AMAZING!!  So far so good! Thank you for all your hard work on this!!

So now I am going around tweaking the core settings trying to get the best picture.  Any tips / advice / best settings?  Basically I am just dealing with "smoothness" around the edges of basically everything.  I have increased the resolution but still seeing it.  I will continue to mess around but figured why not ask the source :).  Again simply awesome!

**Comment by epayson85** [comment]
PS. the sound sync  / audio latency on this is spectacular!!

**Comment by loganmc10** [comment]
Thanks :)

Pretty much the only settings that are going to affect the picture quality are screensize/resolution (the core option). You can try changing the bilinear filtering from standard to 3point if you don't like how the edges look.

All the other settings pretty much just affect what kinds of things get emulated, for instance if you change "Enable depth buffer copy to RDRAM" to "Software" and enabled "Framebuffer Emulation" you should see the lens flare/glare from the lights in Perfect Dark, things like that.

**Comment by loganmc10** [comment]
Ok I've updated the wiki with instructions on how to get this core for Android, ok to close this issue? If you run into any issues please feel free to open a new issue, it would be nice to have more people testing this on Android, since that is where I run into most of the glitches actually.

**Comment by epayson85** [comment]
Well I just posted a link to your core on the hyperspin forums so I bet some folks over there will give it a try.  You are certainly free to close this.  Also oddly enough I got the graphics issue solved. NO IDEA what I did.  I deleted the core specfic config and now it still looks bad when directly booted from retroarch but when booted from hyperspin it looks PERFECT.  I can't seem to figure out where hyperspin has set retroarch to load the config for glupen at this point but I want to figure it out so I can copy the settings and give to others.
